trieste/acquisition/function/entropy.py

```diff
@@ -617,3 +621,231 @@ def __call__(self, x: TensorType) -> TensorType:
             repulsion_weight = 1.0
 
         return repulsion_weight * repulsion
+
+
+MUMBOModelType = TypeVar(
+    "MUMBOModelType", bound=SupportsCovarianceWithTopFidelity, contravariant=True
+)
+""" Type variable bound to :class:`~trieste.models.SupportsCovarianceWithTopFidelity`. """
+
+
+class MUMBO(MinValueEntropySearch[MUMBOModelType]):
+    r"""
+    Builder for the MUlti-task Max-value Bayesian Optimization MUMBO acquisition function modified
+    for objective minimisation. :class:`MinValueEntropySearch` estimates the information in the
+    distribution of the objective minimum that would be gained by evaluating the objective at a
+    given point on a given fidelity level.
+
+    This implementation largely follows :cite:`moss2021mumbo` and samples the objective's minimum
+    :math:`y^*` across a large set of sampled locations via either a Gumbel sampler, an exact
+    Thompson sampler or an approximate random Fourier feature-based Thompson sampler, with the
+    Gumbel sampler being the cheapest but least accurate. Default behavior is to use the
+    exact Thompson sampler.
+    """
+
+    @overload
+    def __init__(
+        self: "MUMBO[SupportsCovarianceWithTopFidelity]",
+        search_space: SearchSpace,
+        num_samples: int = 5,
+        grid_size: int = 1000,
+        min_value_sampler: None = None,
+    ):
+        ...
+
+    @overload
+    def __init__(
+        self: "MUMBO[MUMBOModelType]",
+        search_space: SearchSpace,
+        num_samples: int = 5,
+        grid_size: int = 1000,
+        min_value_sampler: Optional[ThompsonSampler[MUMBOModelType]] = None,
+    ):
+        ...
```

Comment:
I'm not sure I understand what this overloading does?

Reply:
MUMBO is generic in the type of models it supports. This allows you to use it with a min value sampler that only supports certain types of model (second overload), as long as you only use MUMBO with those types of model. However, if you don't specify a min value sampler (first overload) then MUMBO supports any type of model that has covariance with top fidelity.